p2p/server.go

```diff
@@ -319,81 +318,51 @@ func (srv *Server) Peers() []*Peer {
 func (srv *Server) PeerCount() int {
 	var count int
 	select {
-	case srv.peerOp <- func(ps map[enode.ID]*Peer, valNodes map[enode.ID]*valNodeInfo) { count = len(ps) }:
-		<-srv.peerOpDone
-	case <-srv.quit:
-	}
-	return count
-}
-
-// ValPeers returns the validator enodeURLs
-func (srv *Server) ValPeers() []string {
-	var valPeers []string
-	select {
-	case srv.peerOp <- func(ps map[enode.ID]*Peer, valNodes map[enode.ID]*valNodeInfo) {
-		for _, valPeerInfo := range valNodes {
-			valPeers = append(valPeers, valPeerInfo.remoteEnodeURL)
-		}
+	case srv.peerOp <- func(ps map[enode.ID]*Peer) {
+		count = len(ps)
 	}:
 		<-srv.peerOpDone
 	case <-srv.quit:
 	}
-
-	return valPeers
-}
-
-// AddValidatorPeer assigns the given node as a validator node.  It will set it as a static and trusted node.
-func (srv *Server) AddValidatorPeer(node *enode.Node) {
-	select {
-	case srv.addvalidator <- node:
-	case <-srv.quit:
-	}
-}
-
-// RemoveValidatorPeer removes the given node as a validator node.
-func (srv *Server) RemoveValidatorPeer(node *enode.Node) {
-	select {
-	case srv.removevalidator <- node:
-	case <-srv.quit:
-	}
+	return count
 }
 
 // AddPeer connects to the given node and maintains the connection until the
 // server is shut down. If the connection fails for any reason, the server will
 // attempt to reconnect the peer.
-func (srv *Server) AddPeer(node *enode.Node) {
+func (srv *Server) AddPeer(node *enode.Node, purpose string) {
```

Comment:
What do you think about making `purpose` an enum instead of a string?

Reply:
I changed it to be a bitmap based type (with enumerated values).